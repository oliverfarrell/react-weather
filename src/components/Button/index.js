import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";

import { BUTTON_BACKGROUND, BUTTON_BACKGROUND_HOVER, WHITE } from "utils/colors";

import { GLOBAL_FONT_FAMILY, FONT_SIZES, toRem } from "utils";

const StyledButton = styled("button")`
  display: flex;
  align-items: center;
  justify-content: ${props => (props.full ? "center" : "space-between")};

  width: ${props => (props.full ? "100%" : "auto")};
  height: 60px;

  padding: 0 2em;

  font-family: ${GLOBAL_FONT_FAMILY};
  font-size: ${toRem(FONT_SIZES["SMALL"].fontSize)};
  color: ${WHITE};
  background-color: ${BUTTON_BACKGROUND};

  outline: 0;
  border: 0;
  border-radius: 50px;

  cursor: pointer;

  transition: 0.3s background-color ease-in-out;

  &:hover,
  &:active,
  &:focus {
    background-color: ${BUTTON_BACKGROUND_HOVER};
  }
`;

const Button = ({ icon, children, ...props }) => (
  <StyledButton icon={icon} {...props}>
    {children}
    {icon}
  </StyledButton>
);

Button.propTypes = {
  /* Button text */
  children: PropTypes.node,
  /* Button icon */
  icon: PropTypes.node,
  /* Renders a disabled button and removes pointer events if set to true */
  disabled: PropTypes.bool
};

Button.defaultProps = {
  children: "Button",
  icon: undefined,
  disabled: false
};

export default Button;
