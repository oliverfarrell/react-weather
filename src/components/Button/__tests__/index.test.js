import React from "react";
import { mount } from "enzyme";
import { matchers } from "jest-emotion";

import Button from "components/Button";

expect.extend(matchers);

describe("button", () => {
  let wrapper;

  it("renders without crashing", () => {
    wrapper = mount(<Button>Example</Button>);
  });

  it("should render a button", () => {
    expect(wrapper.find("button")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("disabled button", () => {
  let wrapper;

  it("renders without crashing", () => {
    wrapper = mount(<Button disabled />);
  });

  it("should render a button with the disabled attribute", () => {
    expect(wrapper.find("button").prop("disabled")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("full-width button", () => {
  let wrapper;

  it("renders without crashing", () => {
    wrapper = mount(<Button full />);
  });

  it("should have appropriate style rules", () => {
    expect(wrapper).toHaveStyleRule("justify-content", "center");
    expect(wrapper).toHaveStyleRule("width", "100%");
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
