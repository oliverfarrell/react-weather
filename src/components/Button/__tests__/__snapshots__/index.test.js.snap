// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button matches snapshot 1`] = `ReactWrapper {}`;

exports[`disabled button matches snapshot 1`] = `ReactWrapper {}`;

exports[`full-width button matches snapshot 1`] = `ReactWrapper {}`;
